### ISSUE TITLE
`dotnet sln add` and `dotnet sln remove` can fail with exit code 0 and can partially succeed

### ISSUE THREAD

**Issue by ZeroErrors** [dotnet/sdk#18607]
Related to: https://github.com/dotnet/sdk/issues/17686

## Description
The implementation of `dotnet sln add` and `dotnet sln remove` allows them to exit with an exit code of 0 even when failing and also allows them to partially succeed.

For example if you run something like `dotnet sln add File.txt Lib/Lib.csproj` it will succeed adding the `Lib/Lib.csproj` to the solution file and print an message about `File.txt` not being a valid project and then will exit with an exit code of 0.

## Steps to reproduce
```
dotnet new sln
dotnet new console --output src/todo-app
```
Create `some-file.txt`

Fails but exits with exit code 0: `dotnet sln add some-file.txt`
Partially succeeds: `dotnet sln add some-file.txt src/todo-app`

Fails but exits with exit code 0: `dotnet sln remove some-file.txt`
Partially succeeds: `dotnet sln remove some-file.txt src/todo-app`
Running `dotnet sln remove src/todo-app` twice will fail on the second run but exit with exit code 0.

Tested on both main (https://github.com/dotnet/sdk/commit/2c011f2aa7a91a386430233d5797452ca0821ed3) and release 6.0.100-preview.5.21302.13

**Comment by eriawan** [comment]
@ZeroErrors 
I agree, this is closely related to my previously submitted issue of #17686
